File: app/src/main/java/knf/kuma/tv/anime/RelatedPresenter.java
package knf.kuma.tv.anime;

import android.view.ViewGroup;

import androidx.leanback.widget.Presenter;
import knf.kuma.pojos.AnimeObject;
import knf.kuma.tv.cards.RelatedCardView;

public class RelatedPresenter extends Presenter {

    public RelatedPresenter() {
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent) {
        return new ViewHolder(new RelatedCardView(parent.getContext()));
    }

    @Override
    public void onBindViewHolder(ViewHolder viewHolder, Object item) {
        ((RelatedCardView) viewHolder.view).bind((AnimeObject.WebInfo.AnimeRelated) item);
    }

    @Override
    public void onUnbindViewHolder(ViewHolder viewHolder) {

    }
}

File: app/src/main/java/knf/kuma/tv/anime/RelatedPresenter.kt
package knf.kuma.tv.anime

import android.view.ViewGroup

import androidx.leanback.widget.Presenter
import knf.kuma.pojos.AnimeObject
import knf.kuma.tv.cards.RelatedCardView

class RelatedPresenter : Presenter() {

    override fun onCreateViewHolder(parent: ViewGroup): ViewHolder {
        return ViewHolder(RelatedCardView(parent.context))
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, item: Any) {
        (viewHolder.view as RelatedCardView).bind(item as AnimeObject.WebInfo.AnimeRelated)
    }

    override fun onUnbindViewHolder(viewHolder: ViewHolder) {

    }
}
